File: Backend/routes/Admin/analys.js
const express = require('express');
const { Analys } = require('../../db/Models/analys.model');
const { User } = require('../../db/Models/user.model');
const router = express.Router();
const authenticate = require('../controller/authenticateadmin');

// *** add analys  *** //
router.post('/:id', authenticate, (req, res) => {

    User.findById(req.params.id, (err, user) => {

        if (err) throw new Error(err);
        // We create an object containing the data from our analys request
        const newAnalys = new Analys({
            name: req.body.name,
            lastname: req.body.lastname,
            numdemande: req.body.numdemande,
            numdOrder: req.body.numdOrder,
            dateprelv: req.body.dateprelv,
            datenais: req.body.datenais,
            valone: req.body.valone,
            valtwo: req.body.valtwo,
            analysnameone: req.body.analysnameone,
            analysnametwo: req.body.analysnametwo,
            valuselone: req.body.valuselone,
            valuseltwo: req.body.valuseltwo,
            indexnum: req.body.indexnum,
            analysname: req.body.analysname,
            _userId: req.params.id



        });

        // we create our new analys in our database
        Analys.create(newAnalys, (err, analys) => {
            if (err) {
                res.redirect('/');
                throw new Error(err);
            }
            // we save our user with our new data (our new analys).
            analys.save((err) => {
                // return res.redirect(`/users/login/${appoitment.id}`);
                return res.send(newAnalys)
            });
        })
    });

});

// get all analys 

router.get('/', authenticate, (req, res) => {

    Analys.find({}).then((analyss) => {
        res.send(analyss);
    }).catch((e) => {
        res.send(e);
    });

});

/// *** delete analys ***//
router.delete('/:id', authenticate, (req, res) => {
    Analys.findOneAndRemove({
        _id: req.params.id,
    }).then((removeanalysrdoc) => {
        res.send(removeanalysrdoc);
    })


});


module.exports = router;

File: Mobile_telemedicine/lib/Models/Analys/analys.dart
// To parse this JSON data, do
//
//     final analys = analysFromJson(jsonString);

import 'dart:convert';

Analys analysFromJson(String str) => Analys.fromJson(json.decode(str));

String analysToJson(Analys data) => json.encode(data.toJson());

class Analys {
  Analys({
    this.name,
    this.lastname,
    this.numdemande,
    this.numdOrder,
    this.dateprelv,
    this.datenais,
    this.valone,
    this.valtwo,
    this.analysnameone,
    this.analysnametwo,
    this.valuselone,
    this.valuseltwo,
    this.indexnum,
    this.analysname,
  });

  String name;
  String lastname;
  String numdemande;
  String numdOrder;
  String dateprelv;
  String datenais;
  String valone;
  String valtwo;
  String analysnameone;
  String analysnametwo;
  String valuselone;
  String valuseltwo;
  String indexnum;
  String analysname;

  factory Analys.fromJson(Map<String, dynamic> json) => Analys(
        name: json["name"],
        lastname: json["lastname"],
        numdemande: json["numdemande"].toString(),
        numdOrder: json["numdOrder"].toString(),
        dateprelv: json["dateprelv"],
        datenais: json["datenais"],
        valone: json["valone"].toString(),
        valtwo: json["valtwo"].toString(),
        analysnameone: json["analysnameone"],
        analysnametwo: json["analysnametwo"],
        valuselone: json["valuselone"].toString(),
        valuseltwo: json["valuseltwo"].toString(),
        indexnum: json["indexnum"].toString(),
        analysname: json["analysname"],
      );

  Map<String, dynamic> toJson() => {
        "name": name,
        "lastname": lastname,
        "numdemande": numdemande,
        "numdOrder": numdOrder,
        "dateprelv": dateprelv,
        "datenais": datenais,
        "valone": valone,
        "valtwo": valtwo,
        "analysnameone": analysnameone,
        "analysnametwo": analysnametwo,
        "valuselone": valuselone,
        "valuseltwo": valuseltwo,
        "indexnum": indexnum,
        "analysname": analysname,
      };
}
